src/main/java/ch/bfh/bti7081/s2018/green/layouts/UiElement.java

```diff
@@ -0,0 +1,50 @@
+package ch.bfh.bti7081.s2018.green.layouts;
+
+import com.vaadin.data.HasValue;
+import com.vaadin.ui.AbstractComponent;
+import com.vaadin.ui.Component;
+import com.vaadin.ui.CustomLayout;
+
+/*
+* A simple class who contains an element to add to a data-location of a custom layout and location given as string.
+* It allows to creat lists of elements and add them in a foreach or similar to bigger forms
+ */
+public class UiElement {
```

Comment:
Why add UiElement? Please don't add unneeded abstractions

Reply:
It is (or was) historically needed because:

For navigation menu and "buttons" of this kind, we use components like "labels" an no buttons currently (you may not like this decision).

They are derived from a class who can have values (AbstractComponent->AbstracField). But to be able of handling click events and such a button is wrapped around using a Horizontal Layout.
Layouts are not derived fron "AbstractField" what means they do not have a "getValue/setValue" method. So it is needed to have a little different version of "UiElement" called "UiElementHasValue".

But as we disscused, we switch over to using Vaadin-Buttons for Menu-Buttons.
So I will do this till today ca. 20.00pm?